### PR TITLE
[many ports] Add GnuTLS and its dependencies 

### LINKED ISSUE #14746: [New Port Request] gnutls
Library name: GnuTLS

Library description: GnuTLS implements the TLS/SSL (Transport Layer Security aka Secure Sockets Layer) protocol.

Source repository URL: https://github.com/gnutls/gnutls

Project homepage (if different from the source repository):

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by wrobelda**
This PR provides a port for GnuTLS library and its dependencies. 

- Which triplets are supported/not supported? Have you updated the CI baseline?
x64-linux only, for now. Yes.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

**Comment by odskinz488** [comment]
can you please hurry up to port this library faster?!

**Comment by JackBoosY** [comment]
@odskinz488 You can open a new PR, inherit these changes and continue.

**Comment by JackBoosY** [comment]
Pinging @wrobelda for response. Is work still being done for this PR?

**Comment by wrobelda** [comment]
@JackBoosY I intend to finish this work, got held back by migrating to Apple Silicon last month.

**Comment by wrobelda** [comment]
@JackBoosY ready for review. Linux only, for now.

**Comment by wrobelda** [comment]
@JackBoosY I just got it to work on macOS, about to commit those changes

File: ports/libgnutls/CONTROL [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Source: libgnutls
+Version: 3.6.15
+Homepage: https://www.gnutls.org/
+Description: A secure communications library implementing the SSL, TLS and DTLS protocols
+Build-Depends: nettle, gmp, libtasn1, libidn2
+Supports: linux, osx
```
Comment (JackBoosY) [inline ports/libgnutls/CONTROL:6]:
```suggestion
Supports: linux | osx

```

Comment (JackBoosY) [inline ports/libgnutls/CONTROL:6]:
```suggestion
Supports: !windows
```

File: ports/libtasn1/CONTROL [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Source: libtasn1
+Version: 4.16.0
+Homepage: https://www.gnutls.org/
+Description: A secure communications library implementing the SSL, TLS and DTLS protocols
+Supports: linux, osx
```
Comment (JackBoosY) [inline ports/libtasn1/CONTROL:5]:
```suggestion
Supports: !windows
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by wrobelda** [comment]
@JackBoosY the CI build is being skipped for `libgnutls`: "libgnutls:x64-osx:    cascade"

**Comment by JackBoosY** [comment]
@wrobelda  Because libgnutls dependency can't be built on OSX now.

File: versions/l-/libgnutls.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "514174165bb692927e6af1b8d0ba877977760235",
```
Comment (JackBoosY) [inline versions/l-/libgnutls.json:4]:
```suggestion
      "git-tree": "0475cd142427a78431e3cc124b5ec14b189854b7",
```

File: versions/l-/libtasn1.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "40e3f04762264c9abfbd1046a58e0e08768c45dd",
```
Comment (JackBoosY) [inline versions/l-/libtasn1.json:4]:
```suggestion
      "git-tree": "74b4837de3f9ea6b15db7cfcba384a75c91a20d4",
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
This LGTM, thanks for the PR!

**Pull request merged**